teps/0082-workspace-hinting.md

```diff
@@ -0,0 +1,303 @@
+
+### Add a Hinting Field to Workspace Declarations
+
+> The precise name of this field can be iterated on but for now let's assume "profile".
+
+A Workspace Declaration in a Task or Pipeline can include a `profile` field that is a string matching a fixed set of available options:
+- `"cache"` to hint that the workspace will be used as a cache for performance or reproducibility (e.g. a system might require all teams to use a shared `node_modules` directory when compiling their frontends).
+- `"config"` to hint that the workspace is intended to supply some configuration or settings.
+- `"credential"` to hint that the workspace will be used to perform authenticated actions.
+- `"data"` to hint that the contents are arbitrary data either consumed or produced by the Task.
+
+A third-party system can attach its own meaning to each of these
+profiles. A `"credential"` Workspace could be populated from a Secret or
+Secret-like volume. A `"cache"` Workspace might be supplied with a
+long-lived read-only Persistent Volume. A `"data"` Workspace might be
+assumed to require an ephemeral Persistent Volume that lives only
+as long as the PipelineRun. `"configuration"` Workspaces could map
+consistently to `ConfigMaps`. Importantly: these decisions
+are left up to the external / platform. Our own Workflows project may be
+able to utilize these profiles, for example, to make informed choices
+when creating a `PipelineRun` based solely on `Pipeline` YAML, supplied
+list of volumes and set of Secret references.
+
+Here's an example from a `git-clone`-like Task that accepts an optional GitHub deploy key:
+
+```yaml
+workspaces:
+- name: deploy-key
+  readOnly: true
+  optional: true
+  profile: credential
+- name: output
+  profile: data
+```
+
+#### Cons
+- If a Task has multiple "credential" Workspaces, a system still needs
+  to figure out how to bind them. This might automatic (an internal rule
+  like "names must match") or it could require some UI to be exposed
+  asking the user what the correct mapping is.
+
+### Hash-Tags in a Workspace's Description Field
+
+This approach would be entirely ad-hoc: Task authors could include hash tags in their Workspaces' `description` fields. A platform could scan for them and act accordingly. User Interfaces like Hub could be programmed to ignore them or surface them in their own visual component. Here's what Workspaces for a `go-build` Task might look like with these:
+
+```yaml
+workspaces:
+- name: source-code
+  readOnly: true
+  description: "The source of a go program. #data"
+- name: output
+  readOnly: true
+  description: "Compiled binaries will be written here. #data"
+```
+
+#### Pros
+
+- Free-form.
+- The set of recognized hash-tags _could_ be specified and validated by Pipelines (`#cache`, `#config`, `#credential`, `#data`).
+
+#### Cons
+
+- Syntactically different from the "profiles" alternative but otherwise not functionally all that different.
```

Comment:
also i feel like this sets an interesting (imo bad?) precedent for embedding information in descriptions - i guess we kind of do something like this with labels but i feel like at least that's the purpose of labels

Reply:
Good point, I'll include this too.